src/content_scripts/autocorrect.js

```diff
@@ -0,0 +1,351 @@
+					strm += "-";
+				else if (intpart !== 1)
+					strm += intpart;
+
+				strm += constants[i];
+
+				output = true;
+			}
+		}
+	}
+
+	if (!output)
+		strm += anumber;
+
+	return strm;
+}
+
+/**
+ * Get first difference index.
+ *
+ * @param {string} a
+ * @param {string} b
+ * @returns {number}
+ */
+function firstDifferenceIndex(a, b) {
+	if (a === b)
+		return -1;
+	let i = 0;
+	while (a[i] === b[i])
+		++i;
+	return i;
+}
+
+/**
+ * Autocorrect.
+ *
+ * @param {Object} event
+ * @returns {void}
+ */
+function autocorrect(event) {
+	// console.log('keydown', event.key, event.key.length, event.keyCode);
+	if ((event.key.length === 0 || event.key.length === 1 || event.keyCode === 13 || event.key === 'Unidentified') && !event.ctrlKey && !event.metaKey && !event.altKey) {
+		let target = event.target;
+		const caretposition = getCaretPosition(target);
+		if (caretposition) {
+			const value = target.value ? target.value : target.innerText;
+			let deletecount = 0;
+			let insert = value.slice(caretposition - 1, caretposition); // event.key;
+			let output = false;
+			if (quotes && (insert === "'" || insert === '"')) {
+				const prevouschar = value.slice(caretposition < 2 ? 0 : caretposition - 2, caretposition - 1);
+				// White space
+				const re = /^\s*$/;
+				if (insert === "'")
+					insert = re.test(prevouschar) ? '‘' : '’';
+				else if (insert === '"')
+					insert = re.test(prevouschar) ? '“' : '”';
+				deletecount = 1;
+				output = true;
+			}
+			const prevoustext = value.slice(caretposition < (longest + 1) ? 0 : caretposition - (longest + 1), caretposition - 1);
+			let array = symbolpatterns.exec(prevoustext);
+			if (array) {
+				const text = value.slice(caretposition < longest ? 0 : caretposition - longest, caretposition);
+				let aarray = symbolpatterns.exec(text);
+				let aaarray = apatterns.exec(text);
+				if (!aaarray && (!aarray || (caretposition <= longest ? array.index < aarray.index : array.index <= aarray.index))) {
+					insert = autocorrections[array[0]] + (event.keyCode === 13 ? '\n' : insert);
+					deletecount = array[0].length + 1;
+					output = true;
+				}
+			} else {
+				if (autocomplete) {
+					// Emoji Shortcode
+					const re = /:[a-z0-9-+_]+$/;
+					const text = value.slice(caretposition < (longest - 1) ? 0 : caretposition - (longest - 1), caretposition);
+					let array = re.exec(text);
+					if (array) {
+						const aarray = Object.keys(emojiShortcodes).filter((item) => item.indexOf(array[0]) === 0);
+						if (aarray.length === 1 && (array[0].length > 2 || aarray[0].length === 3)) {
+							insert = aarray[0].slice(array[0].length);
+							output = true;
+						}
+					}
+				}
+				if (!output && fracts) {
+					// Numbers
+					const re = /[0-9]+([.][0-9]*)?$/;
+					const prevoustext = value.slice(0, caretposition - 1);
+					let array = re.exec(prevoustext);
+					if (array) {
+						const text = value.slice(0, caretposition);
+						let aarray = re.exec(text);
+						if (!aarray) {
```

Comment:
Suggestions are welcome! Naming things is the [second hardest](https://twitter.com/codinghorror/status/506010907021828096) thing in CS. 🙂

Reply:
Thing is I don't get what the variable is supposed to contain, so possibly you only chose it, because the `array` var is already used.
Actually, naming things as per it's type only is also a bad practise, so let's tackle both arrays. Don't ask yourself what they are (arrays), but what they contain?
So maybe `regexResult`.

Also the other `array` one… hmm…